### PR TITLE
Fix for WFMP-235, Introduce a start-jar goal

### PR DESCRIPTION
**Pull request by jfdenise**
Issue: https://issues.redhat.com/browse/WFMP-235

### CODE REVIEW

**Comment by jamezp** [comment]
I'm on PTO until the 2nd, but I want to think about this. I'll give it a review once I'm back if I don't get bored and review it sooner.  :)

File: plugin/src/main/java/org/wildfly/plugin/server/AbstractServerStartMojo.java [hunk 1]
```diff
@@ -113,190 +78,27 @@ public abstract class AbstractServerStartMojo extends AbstractServerConnection {
...
-    protected MavenRepoManager createMavenRepoManager() throws MojoExecutionException {
-        return new MavenArtifactRepositoryManager(repoSystem, session, repositories);
-    }
-
-    protected ServerContext startServer(final ServerType serverType) throws MojoExecutionException, MojoFailureException {
-        final Log log = getLog();
-        init();
-
+    @Override
+    protected Path getServer() throws MojoExecutionException, MojoFailureException {
```
Comment (jamezp) [inline plugin/src/main/java/org/wildfly/plugin/server/AbstractServerStartMojo.java:95]: I think we should call this `getServerHome()` or `getJBossHome()`.

File: plugin/src/main/java/org/wildfly/plugin/server/StartBootableJarMojo.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+import org.wildfly.plugin.provision.PackageServerMojo;
+import org.wildfly.plugin.tools.bootablejar.BootableJarSupport;
+
+/**
+ * Starts a WildFly Application Server packaged as Bootable JAR.
+ * <p/>
+ * The purpose of this goal is to start a WildFly Application Server packaged as a Bootable JAR for testing during the maven
+ * lifecycle.
+ */
+@Mojo(name = "start-bootable", requiresDependencyResolution = ResolutionScope.RUNTIME)
```
Comment (jamezp) [inline plugin/src/main/java/org/wildfly/plugin/server/StartBootableJarMojo.java:34]: I'm just thinking a bit out loud here. Should the goal name be `start-jar`? Just thinking of something short to type is all, but I'm not sure I like `start-jar` :)

File: plugin/src/main/java/org/wildfly/plugin/server/AbstractServerStartMojo.java (hunk 1 above)
Comment (jfdenise) [reply plugin/src/main/java/org/wildfly/plugin/server/AbstractServerStartMojo.java]: Done thanks.

File: plugin/src/main/java/org/wildfly/plugin/server/StartBootableJarMojo.java (hunk 2 above)
Comment (jfdenise) [reply plugin/src/main/java/org/wildfly/plugin/server/StartBootableJarMojo.java]: I much prefer start-jar, start-bootable is also a bit ambiguous, all server boot... So changed to start-jar.

**Review by jamezp — APPROVED** [review]

**Pull request merged**